src/bin/uhyve.rs

```diff
@@ -392,3 +506,196 @@ fn run_uhyve() -> i32 {
+	///
+	/// Note: Althuogh env_vars and kernel_args can be understood as
+	/// "specific to images", it is assumed that we will be able to
+	/// just append Uhyve-wide and image-specific configurations.
+	#[test]
+	fn test_toml_are_fields_actually_skipped() {
+		let config: Args = toml::from_str(
+			r#"
+			[uhyve]
+			file_mapping = ['foo:bar']
+			config = "/ilikerecursion"
+
+			[guest]
+			kernel = './data/x86_64/hello_c'
+		"#,
+		)
+		.unwrap();
+
+		let file_mapping = config.uhyve.file_mapping;
+		let config_file = config.uhyve.config;
+		let kernel = config.guest.kernel;
+
+		assert!(file_mapping.is_empty());
+		assert!(config_file.is_none());
+		assert!(kernel.to_str().unwrap().is_empty())
+	}
+
+	/// Tests whether TOML merge works as expected.
+	#[test]
+	fn test_toml_merge() {
+		// This is a bit of a workaround and doesn't test the cases that we
+		// deliberately ignore. However, this is covered by other tests.
+		//
+		// Everything with #[merge(skip)] is omitted in the meantime and not
+		// tested explicitly due to triviality.
+		let mut supposed_uhyve_cli_args: Args = toml::from_str(
+			r#"
+			[uhyve]
+			output = "test.txt"
+			stats = true
+			tempdir = "/here"
+			file_isolation = "strict"
+			gdb_port = 1
+
+			[memory]
+			memory_size = '4M'
+			no_aslr = true
+			thp = true
+			ksm = true
+
+			[cpu]
+			cpu_count = 4
+			affinity = [1, 2, 3, 4]
+			pit = true
+
+			[guest]
+			env_vars = ['foo=bar']
+		"#,
+		)
+		.unwrap();
+
+		let config_file: Args = toml::from_str(
+			r#"
+			[uhyve]
+			stats = false
+
+			[memory]
+			no_aslr = false
+			thp = false
+			ksm = false
+
+			[cpu]
+			pit = false
+
+			[guest]
+			env_vars = ['bar=foo']
+		"#,
+		)
+		.unwrap();
+
+		supposed_uhyve_cli_args.merge(config_file);
+
+		// Uhyve parameters
+		assert_eq!(
+			supposed_uhyve_cli_args.uhyve.output,
+			Some("test.txt".into())
+		);
+		assert!(supposed_uhyve_cli_args.uhyve.stats.unwrap());
+		assert_eq!(supposed_uhyve_cli_args.uhyve.tempdir.unwrap(), "/here");
+		#[cfg(target_os = "linux")]
+		assert_eq!(
+			supposed_uhyve_cli_args.uhyve.file_isolation,
+			Some("strict".into())
+		);
+		#[cfg(target_os = "linux")]
+		assert_eq!(supposed_uhyve_cli_args.uhyve.gdb_port.unwrap(), 1);
+
+		// Memory parameters
+		assert_eq!(
+			supposed_uhyve_cli_args.memory.memory_size.unwrap().get(),
+			4000000
+		);
+		assert!(supposed_uhyve_cli_args.memory.no_aslr.unwrap());
+		#[cfg(target_os = "linux")]
+		assert!(supposed_uhyve_cli_args.memory.thp.unwrap());
+		#[cfg(target_os = "linux")]
+		assert!(supposed_uhyve_cli_args.memory.ksm.unwrap());
+
+		// CPU parameters
+		assert_eq!(supposed_uhyve_cli_args.cpu.cpu_count.unwrap().get(), 4);
+		assert!(supposed_uhyve_cli_args.cpu.affinity.is_some());
+		#[cfg(target_os = "linux")]
+		assert!(supposed_uhyve_cli_args.cpu.pit.unwrap());
+
+		// Guest parameters
+		assert_eq!(
+			supposed_uhyve_cli_args.guest.env_vars,
+			vec!["foo=bar", "bar=foo"]
+		);
```

Comment:
Could this be tested more concisely by providing a reference `Args` struct and just comparing `supposed_uhyve_cli_args` with that struct?

Reply:
done in https://github.com/hermit-os/uhyve/pull/1038/commits/a525e44e9dd004cd79ba50f163acc4d27bf8b4ce